framework/src/org/checkerframework/framework/type/DefaultInferredTypesApplier.java

```diff
@@ -110,6 +111,12 @@ private void removePrimaryAnnotationTypeVar(
             TypeMirror typeMirror,
             AnnotationMirror top,
             final AnnotationMirror notInferred) {
+        if (typeMirror.getKind() == TypeKind.WILDCARD) {
```

Comment:
Can you add a bit of javadoc to this method?
The method name suggests that it should only be called on type variables, but now it is also legal to call on wildcards. Would a different name express this better?
What is the relationship between `type` and `typeMirror`? I'm a bit afraid that the `typeMirror` now changes from a wildcard to a type variable, but the `type` stays a wildcard, which would lead the cast on line 129 to fail. Should the loop update both `typeMirror` and `type`? Is `typeMirror` the underlying type of `type`? Some documentation would help understand this.